runner/src/server/plugins/engine/helpers.ts

```diff
@@ -11,9 +11,14 @@ export function proceed(
   nextUrl: string
 ) {
   const returnUrl = request.query.returnUrl;
+  let form_session_identifier = "";
+
+  if (request.query.form_session_identifier) {
+    form_session_identifier = `?form_session_identifier=${request.query.form_session_identifier}`;
+  }
 
   if (typeof returnUrl === "string" && returnUrl.startsWith("/")) {
-    return h.redirect(returnUrl);
+    return h.redirect(`${returnUrl}${form_session_identifier}`);
```

Comment:
What happens if the `returnUrl` already had a query parameter, won't we need to append `form_session_identifier` with a `&` instead of a `?` - we should probably use the built-in URL API to do this.

```suggestion
    const parsedUrl = new URL(returnUrl);
    if (request.query.form_session_identifier) {
        parsedUrl .searchParams.append('form_session_identifier', request.query.form_session_identifier);
    }
    return h.redirect(parsedUrl.toString());
```

Can probably remove the above code too if you do this:

https://github.com/communitiesuk/digital-form-builder/blob/c8db528c0f9ec2229fb702a020787ad9babbbc4e/runner/src/server/plugins/engine/helpers.ts#L14-L18

Reply:
This didn't work because `returnUrl` is a path and not a full url.  So @adamdavies1 reverted in https://github.com/communitiesuk/digital-form-builder/pull/118/commits/82ebe080496b45eab01d26a554d70e7d7e64d452